#!/usr/bin/env python
PACKAGE = "boustrophedon_server"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("outline_layer_count",    int_t,    0, "An Integer parameter", 0,  0, 5)
gen.add("stripe_separation", double_t, 0, "A double parameter",    0.9, 0,   1)
gen.add("intermediary_separation", double_t, 0, "A double parameter",    .0, 0,   10)
gen.add("stripe_angle", double_t, 0, "A double parameter",    0., -3.14,   3.14)
gen.add("repeat_boundary",   bool_t,   0, "A Boolean parameter",  False)
gen.add("outline_clockwise",   bool_t,   0, "A Boolean parameter",  False)
gen.add("skip_outlines",   bool_t,   0, "A Boolean parameter",  True)
gen.add("enable_stripe_angle_orientation",   bool_t,   0, "A Boolean parameter",  True)
gen.add("travel_along_boundary",   bool_t,   0, "A Boolean parameter",  True)
gen.add("return_to_start",   bool_t,   0, "A Boolean parameter",  False)
gen.add("allow_points_outside_boundary",   bool_t,   0, "A Boolean parameter",  True)
gen.add("enable_half_y_turns",   bool_t,   0, "A Boolean parameter",  True)
gen.add("points_per_turn",    int_t,    0, "An Integer parameter", 33,  0, 100)
gen.add("turn_start_offset", double_t, 0, "A double parameter",    .5, 0,   1)


exit(gen.generate(PACKAGE, "boustrophedon_planner_server", "BoustrophedonParameters"))